workloads/imagenet/imagenet_jax/workload.py

```diff
@@ -0,0 +1,290 @@
+        mutable=['batch_stats'])
+      return logits, new_model_state
+    else:
+      logits = self._model.apply(
+        variables,
+        jax.numpy.squeeze(augmented_and_preprocessed_input_batch['image']),
+        train=train,
+        mutable=False)
+      return logits, None
+
+  # Does NOT apply regularization, which is left to the submitter to do in
+  # `update_params`.
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,
+      logits_batch: spec.Tensor) -> spec.Tensor:  # differentiable
+    """Cross Entropy Loss"""
+    one_hot_labels = common_utils.onehot(label_batch,
+                                         num_classes=self.num_classes)
+    xentropy = optax.softmax_cross_entropy(logits=logits_batch,
+                                           labels=one_hot_labels)
+    return jnp.mean(xentropy)
+
+  def compute_metrics(self, logits, labels):
+    loss = self.loss_fn(labels, logits)
+    accuracy = jnp.mean(jnp.argmax(logits, -1) == labels)
+    metrics = {
+      'loss': loss,
+      'accuracy': accuracy,
+    }
+    metrics = lax.pmean(metrics, axis_name='batch')
+    return metrics
+
+  def eval_model(
+      self,
+      params: spec.ParameterTree,
+      model_state: spec.ModelAuxillaryState,
+      rng: spec.RandomState,
+      data_dir: str):
+    """Run a full evaluation of the model."""
+
+    # sync batch statistics across replicas once per epoch
+    model_state = self.sync_batch_stats(model_state)
+
+    epoch_metrics = common_utils.get_metrics(self.epoch_metrics)
+    summary = jax.tree_map(lambda x: x.mean(), epoch_metrics)
+    logging.info('train loss: %.4f, accuracy: %.2f',
+                  summary['loss'], summary['accuracy'] * 100)
+    self.epoch_metrics = []
+    eval_metrics = []
+
+    data_rng, model_rng = jax.random.split(rng, 2)
+    eval_batch_size = self.batch_size
+    num_batches = self.num_eval_examples // eval_batch_size
+    if self._eval_ds is None:
+      self._eval_ds = self._build_dataset(
+        data_rng, split='test', batch_size=eval_batch_size, data_dir=data_dir)
+    eval_iter = iter(self._eval_ds)
+    total_accuracy = 0.
+    eval_step = 1
+    start_time = time.time()
+    accumulated_eval_time = 0
+    for batch in eval_iter:
+      metrics = jax.pmap(self.eval_model_fn,
+                         axis_name='batch',
+                         in_axes=(0, 0, 0, None))(
+                      params,
+                      batch,
+                      model_state,
+                      rng)
+      eval_metrics.append(metrics)
+      total_accuracy += jnp.mean(metrics['accuracy'])
+      eval_step += 1
+      # Check if eval time is over limit
+      current_time = time.time()
+      accumulated_eval_time += current_time - start_time
+      is_time_remaining = (
+        accumulated_eval_time < self.max_allowed_eval_time_sec)
+      if not is_time_remaining:
+        num_batches = eval_step
+        break
+
+    eval_metrics = common_utils.get_metrics(eval_metrics)
```

Comment:
do we need `common_utils.get_metrics`? It grabs the metrics from the 0-th device and places them on the CPU ([code](https://github.com/google/flax/blob/main/flax/training/common_utils.py#L51)), which is all we should need given `eval_metrics` should be a dict of strings -> np.array of shape (num_devices,), I don't think we need the `stack_forest`. You can probably just replace this with just:
```
  eval_metrics = jax.tree_map(lambda x: x[0], eval_metrics)
  eval_metrics = jax.device_get(eval_metrics)
```